src/main/java/com/google/devtools/build/lib/analysis/test/TestActionBuilder.java

```diff
@@ -262,6 +272,26 @@ private TestParams createTestAction(int shards) {
         extraTestEnv.put(CC_CODE_COVERAGE_SCRIPT, collectCcCoverage.getExecPathString());
       }
 
+      if (!instrumentedFiles.getReportedToActualSources().isEmpty()) {
+        Artifact reportedToActualSourcesArtifact =
+            ruleContext.getUniqueDirectoryArtifact("_coverage_helpers","reported_to_actual_sources.txt");
+        String reportedToActualSourcesFileContent =
+            instrumentedFiles.getReportedToActualSources().entrySet().stream()
+                .map(e -> e.getKey() + ":" + e.getValue())
```

Comment:
Parsing will fail when files contain : in their name, won't it? Let's store paths in some escaped format?